types/abci/block.go

```diff
@@ -85,7 +85,7 @@ func ToABCIBlock(block *types.Block) (*cmtypes.Block, error) {
 	}
 
 	val := block.SignedHeader.Validators.Validators[0].Address
-	abciCommit := block.SignedHeader.Commit.ToABCICommit(block.Height(), block.Hash(), val, block.Time())
+	abciCommit := block.SignedHeader.Signature.ToABCICommit(block.Height(), block.Hash(), val, block.Time())
```

Comment:
Jokes aside, all the params passed to `Signature.ToABCICommit` are available directly in `SignedHeader`.
IMHO we should refactor `Signature.ToABCICommit` to something like `SignedHeader.GetABCICommit()`

The only special concern here is `Signature.GetCommitHash` method, but this can be solved (by copying header and updating proposer address before calling `GetABCICommit`).

Reply:
Makes sense. Moved `ToABCICommit` as a util function under `types` package so that it takes `Signature` as an input. I found it cleaner to do that than to "copying header and updating proposer address before calling `GetABCICommit`".